cdap-ui/app/cdap/components/NamespaceDetails/Security/index.js

```diff
@@ -14,47 +14,104 @@
  * the License.
 */
 
-import React from 'react';
+import React, {Component} from 'react';
 import {connect} from 'react-redux';
 import PropTypes from 'prop-types';
 import T from 'i18n-react';
+import AddNamespaceWizard from 'components/CaskWizards/AddNamespace';
+import {getNamespaceProperties} from 'components/NamespaceDetails/store/ActionCreator';
 require('./Security.scss');
 
 const PREFIX = 'features.NamespaceDetails.security';
 
 const mapStateToProps = (state) => {
   return {
+    name: state.name,
+    description: state.description,
+    namespacePrefs: state.namespacePrefs,
+    hdfsRootDirectory: state.hdfsRootDirectory,
+    hbaseNamespaceName: state.hbaseNamespaceName,
+    hiveDatabaseName: state.hiveDatabaseName,
+    schedulerQueueName: state.schedulerQueueName,
     principal: state.principal,
     keytabURI: state.keytabURI
   };
 };
 
-const NamespaceDetailsSecurity = ({principal, keytabURI}) => {
-  return (
-    <div className="namespace-details-security">
-      <div className="namespace-details-section-label">
-        <strong>{T.translate(`${PREFIX}.label`)}</strong>
-      </div>
-      <div className="security-values">
-        <strong>{T.translate(`${PREFIX}.principal`)}</strong>
-        <span title={principal}>
-          {principal || '- -'}
-        </span>
-      </div>
-      <div className="security-values">
-        <strong>{T.translate(`${PREFIX}.keytabURI`)}</strong>
-        <span title={keytabURI}>
-          {keytabURI || '- -'}
-        </span>
-      </div>
-    </div>
-  );
-};
+class NamespaceDetailsSecurity extends Component {
+  state = {
+    modalOpen: false
+  };
 
-NamespaceDetailsSecurity.propTypes = {
-  principal: PropTypes.string,
-  keytabURI: PropTypes.string
-};
+  static propTypes = {
+    name: PropTypes.string,
+    description: PropTypes.string,
+    namespacePrefs: PropTypes.object,
+    hdfsRootDirectory: PropTypes.string,
+    hbaseNamespaceName: PropTypes.string,
+    hiveDatabaseName: PropTypes.string,
+    schedulerQueueName: PropTypes.string,
+    principal: PropTypes.string,
+    keytabURI: PropTypes.string
+  };
+
+  toggleModal = (namespaceEdited) => {
+    if (namespaceEdited) {
+      getNamespaceProperties();
+    }
+    this.setState({
+      modalOpen: !this.state.modalOpen
+    });
+  };
+
+  render() {
+    return (
+      <div className="namespace-details-security">
+        <div className="namespace-details-section-label">
+          <strong>{T.translate(`${PREFIX}.label`)}</strong>
+          {
+            this.props.principal && this.props.keytabURI ?
```

Comment:
Why do we need this check? I can't edit if the namespace initially created without these values?

Reply:
Yes currently we can't edit the keytabURI if the namespace was created without keytabURI and principal properties.